search-books/src/main/java/nix/education/service/response/SolrBookResponseService.java

```diff
@@ -0,0 +1,102 @@
+package nix.education.service.response;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import lombok.SneakyThrows;
+import nix.education.repository.BookRepository;
+import nix.education.service.SavingBookService;
+import nix.education.util.FacetFields;
+import nix.education.util.SymbolConstants;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.response.FacetField;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.client.solrj.response.RangeFacet;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class SolrBookResponseService {
```

Comment:
Should we implement an interface here?

Reply:
We  can, but I guess we don't need it.

Reply:
Implemented interface for ResponseService, SuggestionService and QueryBuilderService